scripts/build-ancillary-repository.sh

```diff
@@ -61,6 +62,48 @@ function build_ancillary_repository() {
 	EOF
 }
 
+function compile_build_info() {
+	local metadir="$1"
+
+	echo "----------------------------------------"
+	echo "APPLIANCE-BUILD ENV"
+	echo "----------------------------------------"
+	echo "APPLIANCE_BUILD_GIT_URL: $APPLIANCE_BUILD_GIT_URL"
+	echo "APPLIANCE_BUILD_GIT_BRANCH: $APPLIANCE_BUILD_GIT_BRANCH"
+	echo "DELPHIX_PACKAGE_MIRROR_MAIN: $DELPHIX_PACKAGE_MIRROR_MAIN"
+	echo "DELPHIX_PACKAGE_MIRROR_SECONDARY: $DELPHIX_PACKAGE_MIRROR_SECONDARY"
+	echo "AWS_S3_OUTPUT: $AWS_S3_OUTPUT"
+	echo "DELPHIX_APPLIANCE_VERSION: $DELPHIX_APPLIANCE_VERSION"
+	echo
+	echo "----------------------------------------"
+	echo "COMPONENTS"
+	echo "----------------------------------------"
+	[[ -f "$metadir/COMPONENTS" ]] && cat "$metadir/COMPONENTS"
```

Comment:
I really dislike how we're so tightly coupling appliance-build to the linux-pkg repository; and this coupling has been only increasing over time.

The original design was for appliance-build to consume arbitrary debian packages and work properly and equally for all consumed debian packages.. but now we seem to be moving to a design where linux-pkg-based packages are handled differently, and perhaps to a design where appliance-build only works with linux-pkg-based packages, rather than arbitrary debian packages.

IMO, appliance-build shouldn't care about the build system of any of our debian packages (e.g. appliance-build shouldn't know that packages were built via linux-pkg or not), but I guess we no longer aspire to such a design...